lexers/janet.lua

```diff
@@ -0,0 +1,172 @@
+-- Copyright 2025 Jason Lenz code@engenforge.com. See LICENSE.
+-- Janet LPeg lexer.
+
+local lexer = lexer
+local P, S = lpeg.P, lpeg.S
+
+local lex = lexer.new(...)
+
+-- Much of the lexical syntax defined below was derived from the following two
+-- Janet documentation links:
+-- https://janet-lang.org/docs/syntax.html
+-- https://janet-lang.org/api/index.html
+
+-- Note that in some cases Janet documentation uses terminology that differs
+-- with Scintillua typical usage. For example, the Janet documentation defines
+-- keywords as symbols that begin with the character ':' and are treated by the
+-- compiler as constants. In this case they were tagged in Scintillua as
+-- constants. Conversely, keywords in Scintillua were defined as built in names
+-- reserved by the janet compiler such as nil, true, do, fn, etc.
+
+lex:add_rule('comment', lex:tag(lexer.COMMENT, lexer.to_eol('#')))
+
+local sq_str = lexer.range('"') + lexer.range('`')
+local dq_str = lexer.range('``')
+local tq_str = lexer.range('```')
+lex:add_rule('string', lex:tag(lexer.STRING, tq_str + dq_str + sq_str))
+
+lex:add_rule('keyword', lex:tag(lexer.KEYWORD, lex:word_match(lexer.KEYWORD)))
+
+lex:add_rule('function', lex:tag(lexer.FUNCTION, lex:word_match(lexer.FUNCTION)))
+
+lex:add_rule('number', lex:tag(lexer.NUMBER, S('-+')^-1 * lexer.digit^1 * (S('._') + lexer.alnum)^0))
+
+local id_ch = S('!@$%^&*-_+=:<>.?') + lexer.alnum
+lex:add_rule('constant', lex:tag(lexer.CONSTANT, P(':') * id_ch^0))
```

Comment:
Is a bare ':' allowed to be a constant? If not, then you probably want to use `id_ch^1`.